core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -866,26 +872,30 @@ public PlanOptimizers(
         builder.add(new StatsRecordingPlanOptimizer(
                 optimizerStats,
                 new PredicatePushDown(metadata, typeOperators, typeAnalyzer, true, false)));
-        builder.add(simplifyOptimizer); // Should be always run after PredicatePushDown
         builder.add(new IterativeOptimizer(
                 metadata,
                 ruleStats,
                 statsCalculator,
                 costCalculator,
-                ImmutableSet.of(new RemoveRedundantTableScanPredicate(metadata, typeOperators))));
+                ImmutableSet.<Rule<?>>builder()
+                        .addAll(simplifyOptimizerRules) // Should be always run after PredicatePushDown
+                        .add(new RemoveRedundantTableScanPredicate(metadata, typeOperators))
+                        .build()));
         builder.add(pushProjectionIntoTableScanOptimizer);
         // Projection pushdown rules may push reducing projections (e.g. dereferences) below filters for potential
         // pushdown into the connectors. Invoke PredicatePushdown and PushPredicateIntoTableScan after this
         // to leverage predicate pushdown on projected columns.
         builder.add(new StatsRecordingPlanOptimizer(optimizerStats, new PredicatePushDown(metadata, typeOperators, typeAnalyzer, true, true)));
         builder.add(new RemoveUnsupportedDynamicFilters(metadata)); // Remove unsupported dynamic filters introduced by PredicatePushdown
-        builder.add(simplifyOptimizer); // Should always run after PredicatePushdown
         builder.add(new IterativeOptimizer(
                 metadata,
                 ruleStats,
                 statsCalculator,
                 costCalculator,
-                ImmutableSet.of(new PushPredicateIntoTableScan(metadata, typeOperators, typeAnalyzer))));
```

Comment:
I think we are just missing `.add(new RemoveRedundantTableScanPredicate(metadata, typeOperators))` here...